src/main/java/org/openstreetmap/atlas/geography/atlas/builder/proto/ProtoAtlasBuilder.java

```diff
@@ -0,0 +1,183 @@
+package org.openstreetmap.atlas.geography.atlas.builder.proto;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.geography.Latitude;
+import org.openstreetmap.atlas.geography.Location;
+import org.openstreetmap.atlas.geography.Longitude;
+import org.openstreetmap.atlas.geography.PolyLine;
+import org.openstreetmap.atlas.geography.atlas.Atlas;
+import org.openstreetmap.atlas.geography.atlas.AtlasMetaData;
+import org.openstreetmap.atlas.geography.atlas.builder.AtlasSize;
+import org.openstreetmap.atlas.geography.atlas.items.Line;
+import org.openstreetmap.atlas.geography.atlas.items.Point;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedAtlas;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedAtlasBuilder;
+import org.openstreetmap.atlas.proto.ProtoAtlasContainer;
+import org.openstreetmap.atlas.proto.ProtoLine;
+import org.openstreetmap.atlas.proto.ProtoLocation;
+import org.openstreetmap.atlas.proto.ProtoPoint;
+import org.openstreetmap.atlas.streaming.resource.Resource;
+import org.openstreetmap.atlas.streaming.resource.WritableResource;
+import org.openstreetmap.atlas.utilities.collections.Maps;
+import org.openstreetmap.atlas.utilities.conversion.ProtoLocationConverter;
+import org.openstreetmap.atlas.utilities.conversion.ProtoTagListConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+
+/**
+ * Build an Atlas from a naive proto serialized file, or write an Atlas to a naive proto serialized
+ * file. At this stage, the builder can only handle Atlases that have tagged Points and Lines.
+ *
+ * @author lcram
+ */
+public class ProtoAtlasBuilder
+{
+    private static final Logger logger = LoggerFactory.getLogger(ProtoAtlasBuilder.class);
+
+    /**
+     * Read a resource in naive ProtoAtlas format into a PackedAtlas.
+     *
+     * @param resource
+     *            the resource in naive ProtoAtlas format
+     * @return the constructed PackedAtlas
+     */
+    public PackedAtlas read(final Resource resource)
+    {
+        ProtoAtlasContainer protoAtlasContainer = null;
+
+        // First, we need to construct the container object from the proto binary
+        try
+        {
+            protoAtlasContainer = ProtoAtlasContainer.parseFrom(resource.readBytesAndClose());
+        }
+        catch (final InvalidProtocolBufferException exception)
+        {
+            throw new CoreException("Error deserializing the ProtoAtlasContainer", exception);
```

Comment:
Here I would pass the name of the resource as well!
```
throw new CoreException("Error deserializing the ProtoAtlasContainer from {}", resource.getName(), exception);
```

Reply:
Awesome I'll add that in.